src/vs/workbench/api/electron-browser/mainThreadEditors.ts

```diff
@@ -170,6 +174,22 @@ export class MainThreadTextEditors implements MainThreadTextEditorsShape {
 		return TPromise.as(null);
 	}
 
+	$trySetLanguageById(id: string, languageId: string): TPromise<void> {
+		let mainThreadEditor = this._documentsAndEditors.getEditor(id);
+		if (!mainThreadEditor) {
+			return TPromise.wrapError(disposed(`TextEditor(${id})`));
+		}
+		let textModel = mainThreadEditor.getModel();
+		let modelService = this._modelService;
+		let modeService = this._modeService;
+		if (!modelService || !modeService) {
```

Comment:
They won't be null/undefined - the injector will explode when a service cannot be found.